### PR TITLE
added test on error when templates have same short name

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Solution
The PR adds additional integration test proving that the error message in  https://github.com/dotnet/templating/issues/3275 and https://github.com/dotnet/templating/issues/4128 is fixed.

https://github.com/dotnet/templating/issues/4135 will allow to resolve the issue better when implemented.

### Checks:
- [x] Added unit tests
- [x] Added `#nullable enable` to all the modified files [?](https://github.com/dotnet/templating/wiki/Contributing#coding-style)

### CODE REVIEW

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs [hunk 1]
```diff
@@ -307,5 +307,33 @@ public Task CanShowWarning_WhenHostDataIsIncorrect()
...
+            string home = TestUtils.CreateTemporaryFolder("Home");
+            string workingDirectory = TestUtils.CreateTemporaryFolder();
+            string templateLocation = Helpers.InstallTestTemplate("Invalid/SameShortName", _log, home, workingDirectory);
+
+            var commandResult = new DotnetNewCommand(_log, "sameshortname")
+                .WithCustomHive(home)
+                .WithWorkingDirectory(workingDirectory)
+                .Execute();
+
+            commandResult
```
Comment (jakrivan) [inline test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs:322]: nit: Why not keep this as one fluent chain? (debugging relict?)

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs [hunk 2]
```diff
@@ -307,5 +307,33 @@ public Task CanShowWarning_WhenHostDataIsIncorrect()
             });
         }
 
+        [Fact]
+        public Task CanShowError_OnTemplatesWithSameShortName()
+        {
+            string home = TestUtils.CreateTemporaryFolder("Home");
+            string workingDirectory = TestUtils.CreateTemporaryFolder();
+            string templateLocation = Helpers.InstallTestTemplate("Invalid/SameShortName", _log, home, workingDirectory);
```
Comment (jakrivan) [inline test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs:315]: Just curious - why do we even allow to install conflicting templates? (or is it already disallowed and this is focused on a case of already installed conflict?)

**Review by jakrivan — APPROVED** [review]

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs (hunk 1 above)
Comment (vlada-shubina) [reply test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs:322]: we need to pass `commandResult` to `Verifier` below. 
Chain will not return it.

File: test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs (hunk 2 above)
Comment (vlada-shubina) [reply test/dotnet-new3.UnitTests/DotnetNewInstantiate.Approval.cs:315]: We considered this before, but as of the moment we consider https://github.com/dotnet/templating/issues/4135 a way to go.

**Comment by vlada-shubina** [comment]
CI has passed, therefore merging.

**Pull request merged**